Fix undefined method in fetch_grant_identifiers

```
2018-11-02T08:34:50-04:00 [ERROR] 72405 : Unexpected error on record <record #15 (data/wos-34839.ndj #15)>
    while executing (to_field "http://vivoweb.org/ontology/core#hasFundingVehicle" at lib/rialto/etl/configs/wos_to_sparql_statements.rb:213)

    Record: {"UID"=>"WOS:000356847800001", "static_data"=>{"summary"=>{"pub_info"=>{"coverdate"=>"JUN 17 2015", "vol"=>16, "journal_oas_gold"=>"Y", "pubyear"=>2015, "sortdate"=>"2015-06-17", "has_abstract"=>"Y", "pubmonth"=>"JUN 17", "pubtype"=>"Journal", "page"=>{"page_count"=>10}}, "names"=>{"count"=>19, "name"=>[{"seq_no"=>1, "role"=>"author", "full_name"=>"Morris, Dylan R.", "addr_no"=>1, "last_name"=>"Morris", "display_name"=>"Morris, Dylan R.", "wos_standard"=>"Morris, DR", "daisng_id"=>2180566, "first_name"=>"Dylan R."}, {"seq_no"=>2, "role"=>"author", "full_name"=>"Cunningham, Margaret A.", "addr_no"=>2, "last_name"=>"Cunningham", "display_name"=>"Cunningham, Margaret A.", "wos_standard"=>"Cunningham, MA", "daisng_id"=>3752222, "first_name"=>"Margaret A."}, {"seq_no"=>3, "role"=>"author", "full_name"=>"Ahimastos, Anna A.", "addr_no"=>"3 4", "last_name"=>"Ahimastos", "display_name"=>"Ahimastos, Anna A.", "wos_standard"=>"Ahimastos, AA", "daisng_id"=>1273296, "first_name"=>"Anna A."}, {"seq_no"=>4, "role"=>"author", "full_name"=>"Kingwell, Bronwyn A.", "addr_no"=>"3 4", "last_name"=>"Kingwell", "display_name"=>"Kingwell, Bronwyn A.", "wos_standard"=>"Kingwell, BA", "daisng_id"=>62380, "first_name"=>"Bronwyn A."}, {"seq_no"=>5, "role"=>"author", "full_name"=>"Pappas, Elise", "addr_no"=>1, "last_name"=>"Pappas", "display_name"=>"Pappas, Elise", "wos_standard"=>"Pappas, E", "daisng_id"=>4043963, "first_name"=>"Elise"}, {"seq_no"=>6, "role"=>"author", "full_name"=>"Bourke, Michael", "addr_no"=>5, "last_name"=>"Bourke", "display_name"=>"Bourke, Michael", "wos_standard"=>"Bourke, M", "daisng_id"=>25679, "first_name"=>"Michael"}, {"seq_no"=>7, "role"=>"author", "full_name"=>"Reid, Christopher M.", "addr_no"=>6, "last_name"=>"Reid", "display_name"=>"Reid, Christopher M.", "wos_standard"=>"Reid, CM", "daisng_id"=>189946, "first_name"=>"Christopher M."}, {"seq_no"=>8, "role"=>"author", "full_name"=>"Stijnen, Theo", "addr_no"=>7, "last_name"=>"Stijnen", "display_name"=>"Stijnen, Theo", "wos_standard"=>"Stijnen, T", "daisng_id"=>24285, "first_name"=>"Theo"}, {"seq_no"=>9, "role"=>"author", "full_name"=>"Dalman, Ronald L.", "addr_no"=>8, "last_name"=>"Dalman", "display_name"=>"Dalman, Ronald L.", "wos_standard"=>"Dalman, RL", "daisng_id"=>74005, "first_name"=>"Ronald L."}, {"seq_no"=>10, "role"=>"author", "full_name"=>"Aalami, Oliver O.", "addr_no"=>8, "last_name"=>"Aalami", "display_name"=>"Aalami, Oliver O.", "wos_standard"=>"Aalami, OO", "daisng_id"=>919385, "first_name"=>"Oliver O."}, {"seq_no"=>11, "role"=>"author", "full_name"=>"Lindeman, Jan H.", "addr_no"=>7, "last_name"=>"Lindeman", "display_name"=>"Lindeman, Jan H.", "wos_standard"=>"Lindeman, JH", "daisng_id"=>97383, "first_name"=>"Jan H."}, {"seq_no"=>12, "role"=>"author", "full_name"=>"Norman, Paul E.", "addr_no"=>9, "last_name"=>"Norman", "display_name"=>"Norman, Paul E.", "wos_standard"=>"Norman, PE", "daisng_id"=>29698, "first_name"=>"Paul E."}, {"seq_no"=>13, "role"=>"author", "full_name"=>"Walker, Philip J.", "addr_no"=>"10 11 12", "last_name"=>"Walker", "display_name"=>"Walker, Philip J.", "wos_standard"=>"Walker, PJ", "daisng_id"=>248907, "first_name"=>"Philip J."}, {"seq_no"=>14, "role"=>"author", "full_name"=>"Fitridge, Robert", "addr_no"=>13, "last_name"=>"Fitridge", "display_name"=>"Fitridge, Robert", "wos_standard"=>"Fitridge, R", "daisng_id"=>360408, "first_name"=>"Robert"}, {"seq_no"=>15, "role"=>"author", "full_name"=>"Bourke, Bernie", "addr_no"=>5, "last_name"=>"Bourke", "display_name"=>"Bourke, Bernie", "wos_standard"=>"Bourke, B", "daisng_id"=>1374622, "first_name"=>"Bernie"}, {"seq_no"=>16, "orcid_id"=>"0000-0001-7832-2294", "role"=>"author", "full_name"=>"Dear, Anthony E.", "addr_no"=>14, "last_name"=>"Dear", "display_name"=>"Dear, Anthony E.", "wos_standard"=>"Dear, AE", "daisng_id"=>604898, "first_name"=>"Anthony E."}, {"seq_no"=>17, "role"=>"author", "full_name"=>"Pinchbeck, Jenna", "addr_no"=>1, "last_name"=>"Pinchbeck", "display_name"=>"Pinchbeck, Jenna", "wos_standard"=>"Pinchbeck, J", "daisng_id"=>5055189, "first_name"=>"Jenna"}, {"seq_no"=>18, "role"=>"author", "full_name"=>"Jaeggi, Rene", "addr_no"=>1, "last_name"=>"Jaeggi", "display_name"=>"Jaeggi, Rene", "wos_standard"=>"Jaeggi, R", "daisng_id"=>5626222, "first_name"=>"Rene"}, {"seq_no"=>19, "role"=>"author", "full_name"=>"Golledge, Jonathan", "addr_no"=>"1 15", "last_name"=>"Golledge", "display_name"=>"Golledge, Jonathan", "wos_standard"=>"Golledge, J", "daisng_id"=>31770, "first_name"=>"Jonathan"}]}, "doctypes"=>{"doctype"=>"Article", "count"=>1}, "publishers"=>{"publisher"=>{"names"=>{"count"=>1, "name"=>{"seq_no"=>1, "role"=>"publisher", "full_name"=>"BIOMED CENTRAL LTD", "addr_no"=>1, "display_name"=>"BIOMED CENTRAL LTD"}}, "address_spec"=>{"city"=>"LONDON", "addr_no"=>1, "full_address"=>"236 GRAYS INN RD, FLOOR 6, LONDON WC1X 8HL, ENGLAND"}}}, "EWUID"=>{"WUID"=>{"coll_id"=>"WOS"}, "edition"=>{"value"=>"WOS.SCI"}}, "titles"=>{"count"=>6, "title"=>[{"type"=>"source", "content"=>"TRIALS"}, {"type"=>"source_abbrev", "content"=>"TRIALS"}, {"type"=>"abbrev_iso", "content"=>"Trials"}, {"type"=>"abbrev_11", "content"=>"TRIALS"}, {"type"=>"abbrev_29", "content"=>"TRIALS"}, {"type"=>"item", "content"=>"TElmisartan in the management of abDominal aortic aneurYsm (TEDY): The study protocol for a randomized controlled trial"}]}}, "item"=>{"xsi:type"=>"itemType_wos", "coll_id"=>"WOS", "ids"=>{"avail"=>"Y", "content"=>"CL3KC"}, "xmlns:xsi"=>"http://www.w3.org/2001/XMLSchema-instance", "bib_pagecount"=>{"type"=>"Journal", "content"=>10}, "keywords_plus"=>{"count"=>10, "keyword"=>["CONVERTING ENZYME-INHIBITORS", "ACTIVATED-RECEPTOR-GAMMA", "QUALITY-OF-LIFE", "ANGIOTENSIN-II", "MOUSE MODEL", "DOUBLE-BLIND", "GROWTH-RATE", "SURVEILLANCE", "PROGRESSION", "RISK"]}, "bib_id"=>"16: - JUN 17 2015"}, "fullrecord_metadata"=>{"addresses"=>{"count"=>15, "address_name"=>[{"names"=>{"count"=>5, "name"=>[{"seq_no"=>1, "role"=>"author", "full_name"=>"Morris, Dylan R.", "addr_no"=>1, "last_name"=>"Morris", "display_name"=>"Morris, Dylan R.", "wos_standard"=>"Morris, DR", "daisng_id"=>2180566, "first_name"=>"Dylan R."}, {"seq_no"=>5, "role"=>"author", "full_name"=>"Pappas, Elise", "addr_no"=>1, "last_name"=>"Pappas", "display_name"=>"Pappas, Elise", "wos_standard"=>"Pappas, E", "daisng_id"=>4043963, "first_name"=>"Elise"}, {"seq_no"=>17, "role"=>"author", "full_name"=>"Pinchbeck, Jenna", "addr_no"=>1, "last_name"=>"Pinchbeck", "display_name"=>"Pinchbeck, Jenna", "wos_standard"=>"Pinchbeck, J", "daisng_id"=>5055189, "first_name"=>"Jenna"}, {"seq_no"=>18, "role"=>"author", "full_name"=>"Jaeggi, Rene", "addr_no"=>1, "last_name"=>"Jaeggi", "display_name"=>"Jaeggi, Rene", "wos_standard"=>"Jaeggi, R", "daisng_id"=>5626222, "first_name"=>"Rene"}, {"seq_no"=>19, "role"=>"author", "full_name"=>"Golledge, Jonathan", "addr_no"=>1, "last_name"=>"Golledge", "display_name"=>"Golledge, Jonathan", "wos_standard"=>"Golledge, J", "daisng_id"=>31770, "first_name"=>"Jonathan"}]}, "address_spec"=>{"zip"=>{"location"=>"AP", "content"=>4811}, "country"=>"Australia", "city"=>"Townsville", "addr_no"=>1, "organizations"=>{"organization"=>["James Cook Univ", {"pref"=>"Y", "content"=>"James Cook University"}], "count"=>2}, "full_address"=>"James Cook Univ, Sch Med & Dent, Queensland Res Ctr Peripheral Vasc Dis, Townsville, Qld 4811, Australia", "state"=>"Qld", "suborganizations"=>{"count"=>2, "suborganization"=>["Sch Med & Dent", "Queensland Res Ctr Peripheral Vasc Dis"]}}}, {"names"=>{"count"=>1, "name"=>{"seq_no"=>2, "role"=>"author", "full_name"=>"Cunningham, Margaret A.", "addr_no"=>2, "last_name"=>"Cunningham", "display_name"=>"Cunningham, Margaret A.", "wos_standard"=>"Cunningham, MA", "daisng_id"=>3752222, "first_name"=>"Margaret A."}}, "address_spec"=>{"zip"=>{"location"=>"AC", "content"=>"FK9 4LA"}, "country"=>"Scotland", "city"=>"Stirling", "addr_no"=>2, "organizations"=>{"organization"=>["Univ Stirling", {"pref"=>"Y", "content"=>"University of Stirling"}], "count"=>2}, "full_address"=>"Univ Stirling, Dept Psychol, Stirling FK9 4LA, Scotland", "suborganizations"=>{"count"=>1, "suborganization"=>"Dept Psychol"}}}, {"names"=>{"count"=>2, "name"=>[{"seq_no"=>3, "role"=>"author", "full_name"=>"Ahimastos, Anna A.", "addr_no"=>3, "last_name"=>"Ahimastos", "display_name"=>"Ahimastos, Anna A.", "wos_standard"=>"Ahimastos, AA", "daisng_id"=>1273296, "first_name"=>"Anna A."}, {"seq_no"=>4, "role"=>"author", "full_name"=>"Kingwell, Bronwyn A.", "addr_no"=>3, "last_name"=>"Kingwell", "display_name"=>"Kingwell, Bronwyn A.", "wos_standard"=>"Kingwell, BA", "daisng_id"=>62380, "first_name"=>"Bronwyn A."}]}, "address_spec"=>{"country"=>"Australia", "city"=>"Melbourne", "addr_no"=>3, "organizations"=>{"organization"=>["Alfred Hosp Melbourne", {"pref"=>"Y", "content"=>"Baker Heart and Diabetes Institute"}], "count"=>2}, "full_address"=>"Alfred Hosp Melbourne, Baker IDI Heart & Diabet Inst, Melbourne, Vic, Australia", "state"=>"Vic", "suborganizations"=>{"count"=>1, "suborganization"=>"Baker IDI Heart & Diabet Inst"}}}, {"names"=>{"count"=>2, "name"=>[{"seq_no"=>3, "role"=>"author", "full_name"=>"Ahimastos, Anna A.", "addr_no"=>4, "last_name"=>"Ahimastos", "display_name"=>"Ahimastos, Anna A.", "wos_standard"=>"Ahimastos, AA", "daisng_id"=>1273296, "first_name"=>"Anna A."}, {"seq_no"=>4, "role"=>"author", "full_name"=>"Kingwell, Bronwyn A.", "addr_no"=>4, "last_name"=>"Kingwell", "display_name"=>"Kingwell, Bronwyn A.", "wos_standard"=>"Kingwell, BA", "daisng_id"=>62380, "first_name"=>"Bronwyn A."}]}, "address_spec"=>{"country"=>"Australia", "city"=>"Melbourne", "addr_no"=>4, "organizations"=>{"organization"=>"Alfred Hosp Melbourne", "count"=>1}, "full_address"=>"Alfred Hosp Melbourne, Dept Cardiovasc Med, Melbourne, Vic, Australia", "state"=>"Vic", "suborganizations"=>{"count"=>1, "suborganization"=>"Dept Cardiovasc Med"}}}, {"names"=>{"count"=>2, "name"=>[{"seq_no"=>6, "role"=>"author", "full_name"=>"Bourke, Michael", "addr_no"=>5, "last_name"=>"Bourke", "display_name"=>"Bourke, Michael", "wos_standard"=>"Bourke, M", "daisng_id"=>25679, "first_name"=>"Michael"}, {"seq_no"=>15, "role"=>"author", "full_name"=>"Bourke, Bernie", "addr_no"=>5, "last_name"=>"Bourke", "display_name"=>"Bourke, Bernie", "wos_standard"=>"Bourke, B", "daisng_id"=>1374622, "first_name"=>"Bernie"}]}, "address_spec"=>{"country"=>"Australia", "city"=>"Gosford", "addr_no"=>5, "organizations"=>{"organization"=>"Gosford Vasc Serv", "count"=>1}, "full_address"=>"Gosford Vasc Serv, Gosford, NSW, Australia", "state"=>"NSW"}}, {"names"=>{"count"=>1, "name"=>{"seq_no"=>7, "role"=>"author", "full_name"=>"Reid, Christopher M.", "addr_no"=>6, "last_name"=>"Reid", "display_name"=>"Reid, Christopher M.", "wos_standard"=>"Reid, CM", "daisng_id"=>189946, "first_name"=>"Christopher M."}}, "address_spec"=>{"zip"=>{"location"=>"AP", "content"=>3181}, "country"=>"Australia", "city"=>"Melbourne", "addr_no"=>6, "organizations"=>{"organization"=>["Monash Univ", {"pref"=>"Y", "content"=>"Monash University"}, {"pref"=>"Y", "content"=>"Florey Institute of Neuroscience & Mental Health"}], "count"=>3}, "full_address"=>"Monash Univ, Alfred Hosp, Dept Epidemiol & Prevent Med, Ctr Cardiovasc Res & Educ Therapeut, Melbourne, Vic 3181, Australia", "state"=>"Vic", "suborganizations"=>{"count"=>3, "suborganization"=>["Alfred Hosp", "Dept Epidemiol & Prevent Med", "Ctr Cardiovasc Res & Educ Therapeut"]}}}, {"names"=>{"count"=>2, "name"=>[{"seq_no"=>8, "role"=>"author", "full_name"=>"Stijnen, Theo", "addr_no"=>7, "last_name"=>"Stijnen", "display_name"=>"Stijnen, Theo", "wos_standard"=>"Stijnen, T", "daisng_id"=>24285, "first_name"=>"Theo"}, {"seq_no"=>11, "role"=>"author", "full_name"=>"Lindeman, Jan H.", "addr_no"=>7, "last_name"=>"Lindeman", "display_name"=>"Lindeman, Jan H.", "wos_standard"=>"Lindeman, JH", "daisng_id"=>97383, "first_name"=>"Jan H."}]}, "address_spec"=>{"country"=>"Netherlands", "city"=>"Leiden", "addr_no"=>7, "organizations"=>{"organization"=>["Leiden Univ", {"pref"=>"Y", "content"=>"Leiden University"}], "count"=>2}, "full_address"=>"Leiden Univ, Med Ctr, Leiden, Netherlands", "suborganizations"=>{"count"=>1, "suborganization"=>"Med Ctr"}}}, {"names"=>{"count"=>2, "name"=>[{"seq_no"=>9, "role"=>"author", "full_name"=>"Dalman, Ronald L.", "addr_no"=>8, "last_name"=>"Dalman", "display_name"=>"Dalman, Ronald L.", "wos_standard"=>"Dalman, RL", "daisng_id"=>74005, "first_name"=>"Ronald L."}, {"seq_no"=>10, "role"=>"author", "full_name"=>"Aalami, Oliver O.", "addr_no"=>8, "last_name"=>"Aalami", "display_name"=>"Aalami, Oliver O.", "wos_standard"=>"Aalami, OO", "daisng_id"=>919385, "first_name"=>"Oliver O."}]}, "address_spec"=>{"zip"=>{"location"=>"AP", "content"=>94305}, "country"=>"USA", "city"=>"Stanford", "addr_no"=>8, "organizations"=>{"organization"=>["Stanford Univ", {"pref"=>"Y", "content"=>"Stanford University"}], "count"=>2}, "full_address"=>"Stanford Univ, Sch Med, Dept Surg, Div Vasc Surg, Stanford, CA 94305 USA", "state"=>"CA", "suborganizations"=>{"count"=>3, "suborganization"=>["Sch Med", "Dept Surg", "Div Vasc Surg"]}}}, {"names"=>{"count"=>1, "name"=>{"seq_no"=>12, "role"=>"author", "full_name"=>"Norman, Paul E.", "addr_no"=>9, "last_name"=>"Norman", "display_name"=>"Norman, Paul E.", "wos_standard"=>"Norman, PE", "daisng_id"=>29698, "first_name"=>"Paul E."}}, "address_spec"=>{"zip"=>{"location"=>"AP", "content"=>6009}, "country"=>"Australia", "city"=>"Perth", "addr_no"=>9, "organizations"=>{"organization"=>["Univ Western Australia", {"pref"=>"Y", "content"=>"University of Western Australia"}], "count"=>2}, "full_address"=>"Univ Western Australia, Sch Surg, Perth, WA 6009, Australia", "state"=>"WA", "suborganizations"=>{"count"=>1, "suborganization"=>"Sch Surg"}}}, {"names"=>{"count"=>1, "name"=>{"seq_no"=>13, "role"=>"author", "full_name"=>"Walker, Philip J.", "addr_no"=>10, "last_name"=>"Walker", "display_name"=>"Walker, Philip J.", "wos_standard"=>"Walker, PJ", "daisng_id"=>248907, "first_name"=>"Philip J."}}, "address_spec"=>{"country"=>"Australia", "city"=>"Brisbane", "addr_no"=>10, "organizations"=>{"organization"=>["Univ Queensland", {"pref"=>"Y", "content"=>"University of Queensland"}], "count"=>2}, "full_address"=>"Univ Queensland, Sch Med, Discipline Surg, Brisbane, Qld, Australia", "state"=>"Qld", "suborganizations"=>{"count"=>2, "suborganization"=>["Sch Med", "Discipline Surg"]}}}, {"names"=>{"count"=>1, "name"=>{"seq_no"=>13, "role"=>"author", "full_name"=>"Walker, Philip J.", "addr_no"=>11, "last_name"=>"Walker", "display_name"=>"Walker, Philip J.", "wos_standard"=>"Walker, PJ", "daisng_id"=>248907, "first_name"=>"Philip J."}}, "address_spec"=>{"country"=>"Australia", "city"=>"Brisbane", "addr_no"=>11, "organizations"=>{"organization"=>["Univ Queensland", {"pref"=>"Y", "content"=>"University of Queensland"}], "count"=>2}, "full_address"=>"Univ Queensland, Clin Res Ctr, Brisbane, Qld, Australia", "state"=>"Qld", "suborganizations"=>{"count"=>1, "suborganization"=>"Clin Res Ctr"}}}, {"names"=>{"count"=>1, "name"=>{"seq_no"=>13, "role"=>"author", "full_name"=>"Walker, Philip J.", "addr_no"=>12, "last_name"=>"Walker", "display_name"=>"Walker, Philip J.", "wos_standard"=>"Walker, PJ", "daisng_id"=>248907, "first_name"=>"Philip J."}}, "address_spec"=>{"country"=>"Australia", "city"=>"Brisbane", "addr_no"=>12, "organizations"=>{"organization"=>["Univ Queensland", {"pref"=>"Y", "content"=>"University of Queensland"}, {"pref"=>"Y", "content"=>"Royal Brisbane & Women's Hospital"}], "count"=>3}, "full_address"=>"Univ Queensland, Dept Vasc Surg, Royal Brisbane & Womens Hosp, Brisbane, Qld, Australia", "state"=>"Qld", "suborganizations"=>{"count"=>2, "suborganization"=>["Dept Vasc Surg", "Royal Brisbane & Womens Hosp"]}}}, {"names"=>{"count"=>1, "name"=>{"seq_no"=>14, "role"=>"author", "full_name"=>"Fitridge, Robert", "addr_no"=>13, "last_name"=>"Fitridge", "display_name"=>"Fitridge, Robert", "wos_standard"=>"Fitridge, R", "daisng_id"=>360408, "first_name"=>"Robert"}}, "address_spec"=>{"country"=>"Australia", "city"=>"Adelaide", "addr_no"=>13, "organizations"=>{"organization"=>["Univ Adelaide", {"pref"=>"Y", "content"=>"University of Adelaide"}], "count"=>2}, "full_address"=>"Univ Adelaide, Queen Elizabeth Hosp, Dept Surg, Adelaide, SA, Australia", "state"=>"SA", "suborganizations"=>{"count"=>2, "suborganization"=>["Queen Elizabeth Hosp", "Dept Surg"]}}}, {"names"=>{"count"=>1, "name"=>{"seq_no"=>16, "role"=>"author", "full_name"=>"Dear, Anthony E.", "addr_no"=>14, "last_name"=>"Dear", "display_name"=>"Dear, Anthony E.", "wos_standard"=>"Dear, AE", "daisng_id"=>604898, "first_name"=>"Anthony E."}}, "address_spec"=>{"zip"=>{"location"=>"AP", "content"=>3004}, "country"=>"Australia", "city"=>"Melbourne", "addr_no"=>14, "organizations"=>{"organization"=>["Monash Univ", {"pref"=>"Y", "content"=>"Monash University"}], "count"=>2}, "full_address"=>"Monash Univ, Dept Med, Eastern Hlth Clin Sch, Melbourne, Vic 3004, Australia", "state"=>"Vic", "suborganizations"=>{"count"=>2, "suborganization"=>["Dept Med", "Eastern Hlth Clin Sch"]}}}, {"names"=>{"count"=>1, "name"=>{"seq_no"=>19, "role"=>"author", "full_name"=>"Golledge, Jonathan", "addr_no"=>15, "last_name"=>"Golledge", "display_name"=>"Golledge, Jonathan", "wos_standard"=>"Golledge, J", "daisng_id"=>31770, "first_name"=>"Jonathan"}}, "address_spec"=>{"country"=>"Australia", "city"=>"Townsville", "addr_no"=>15, "organizations"=>{"organization"=>"Townsville Hosp", "count"=>1}, "full_address"=>"Townsville Hosp, Dept Vasc & Endovasc Surg, Townsville, Qld, Australia", "state"=>"Qld", "suborganizations"=>{"count"=>1, "suborganization"=>"Dept Vasc & Endovasc Surg"}}}]}, "category_info"=>{"subheadings"=>{"count"=>1, "subheading"=>"Life Sciences & Biomedicine"}, "subjects"=>{"subject"=>[{"ascatype"=>"traditional", "code"=>"QA", "content"=>"Medicine, Research & Experimental"}, {"ascatype"=>"extended", "content"=>"Research & Experimental Medicine"}], "count"=>2}, "headings"=>{"heading"=>"Science & Technology", "count"=>1}}, "normalized_languages"=>{"count"=>1, "language"=>{"type"=>"primary", "content"=>"English"}}, "languages"=>{"count"=>1, "language"=>{"type"=>"primary", "content"=>"English"}}, "keywords"=>{"count"=>4, "keyword"=>["Abdominal aortic aneurysm", "Trial", "Angiotensin", "Telmisartan"]}, "refs"=>{"count"=>60}, "abstracts"=>{"count"=>1, "abstract"=>{"abstract_text"=>{"p"=>["Background: Experimental studies suggest that angiotensin II plays a central role in the pathogenesis of abdominal aortic aneurysm. This trial aims to evaluate the efficacy of the angiotensin receptor blocker telmisartan in limiting the progression of abdominal aortic aneurysm.", "Methods/Design: Telmisartan in the management of abdominal aortic aneurysm (TEDY) is a multicentre, parallel-design, randomised, double-blind, placebo-controlled trial with an intention-to-treat analysis. We aim to randomly assign 300 participants with small abdominal aortic aneurysm to either 40 mg of telmisartan or identical placebo and follow patients over 2 years. The primary endpoint will be abdominal aortic aneurysm growth as measured by 1) maximum infra-renal aortic volume on computed tomographic angiography, 2) maximum orthogonal diameter on computed tomographic angiography, and 3) maximum diameter on ultrasound. Secondary endpoints include change in resting brachial blood pressure, abdominal aortic aneurysm biomarker profile and health-related quality of life. TEDY is an international collaboration conducted from major vascular centres in Australia, the United States and the Netherlands.", "Discussion: Currently, no medication has been convincingly demonstrated to limit abdominal aortic aneurysm progression. TEDY will examine the potential of a promising treatment strategy for patients with small abdominal aortic aneurysms."], "count"=>3}}}, "fund_ack"=>{"grants"=>{"count"=>5, "grant"=>[{"grant_ids"=>{"grant_id"=>[1022752, 526604], "count"=>2}, "grant_agency"=>"National Health and Medical Research Council of Australia"}, {"grant_agency"=>"British United Provident Association Foundation"}, {"grant_ids"=>{"grant_id"=>1000967, "count"=>1}, "grant_agency"=>"Centre of Research Excellence grant from the National Health and Medical Research Council of Australia"}, {"grant_ids"=>{"grant_id"=>1019921, "count"=>1}, "grant_agency"=>"National Health and Medical Research Council, Australia"}, {"grant_agency"=>"Queensland Government"}]}, "fund_text"=>{"p"=>"This trial is funded by project grants from the National Health and Medical Research Council of Australia (1022752) and the British United Provident Association Foundation and a Centre of Research Excellence grant from the National Health and Medical Research Council of Australia (1000967). JG holds a Practitioner Fellowship from the National Health and Medical Research Council, Australia (1019921) and a Senior Clinical Research Fellowship from the Queensland Government. BAK holds a Principal Research Fellowship from the National Health and Medical Research Council of Australia (526604). The sponsors had no role in the design and conduct of the study; collection, management, analysis, and interpretation of the data; or preparation, review, or approval of the manuscript."}}, "normalized_doctypes"=>{"doctype"=>"Article", "count"=>1}}, "contributors"=>{"contributor"=>{"name"=>{"seq_no"=>1, "orcid_id"=>"0000-0001-7832-2294", "role"=>"researcher_id", "full_name"=>"Dear, Anthony", "last_name"=>"Dear", "display_name"=>"Dear, Anthony", "first_name"=>"Anthony"}}, "count"=>1}}, "r_id_disclaimer"=>"ResearcherID data provided by Clarivate Analytics", "dynamic_data"=>{"citation_related"=>{"tc_list"=>{"silo_tc"=>{"coll_id"=>"WOS", "local_count"=>15}}}, "cluster_related"=>{"identifiers"=>{"identifier"=>[{"type"=>"issn", "value"=>"1745-6215"}, {"type"=>"art_no", "value"=>"ARTN 274"}, {"type"=>"doi", "value"=>"10.1186/s13063-015-0793-z"}, {"type"=>"pmid", "value"=>"MEDLINE:26081587"}]}}}}
    Exception: NoMethodError: undefined method `empty?' for 1022752:Integer
    lib/rialto/etl/configs/wos_to_sparql_statements.rb:66:in `block in fetch_grant_identifiers'

2018-11-02T08:34:50-04:00 [DEBUG] 72405 : #processing mapper threadpool shutdown complete.
Traceback (most recent call last):
	19: from /Users/jlittman/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:319:in `block in create_worker'
	18: from /Users/jlittman/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:319:in `catch'
	17: from /Users/jlittman/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:320:in `block (2 levels) in create_worker'
	16: from /Users/jlittman/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:320:in `loop'
	15: from /Users/jlittman/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:337:in `block (3 levels) in create_worker'
	14: from /Users/jlittman/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:348:in `run_task'
	13: from /Users/jlittman/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/traject-3.0.0/lib/traject/thread_pool.rb:123:in `block in maybe_in_thread_pool'
	12: from /Users/jlittman/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/traject-3.0.0/lib/traject/indexer.rb:554:in `block (3 levels) in process'
	11: from /Users/jlittman/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/traject-3.0.0/lib/traject/indexer.rb:429:in `map_to_context!'
	10: from /Users/jlittman/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/traject-3.0.0/lib/traject/indexer.rb:429:in `each'
	 9: from /Users/jlittman/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/traject-3.0.0/lib/traject/indexer.rb:435:in `block in map_to_context!'
	 8: from /Users/jlittman/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/traject-3.0.0/lib/traject/indexer.rb:476:in `handle_mapping_errors'
	 7: from /Users/jlittman/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/traject-3.0.0/lib/traject/indexer.rb:436:in `block (2 levels) in map_to_context!'
	 6: from /Users/jlittman/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/traject-3.0.0/lib/traject/indexer/step.rb:135:in `execute'
	 5: from /Users/jlittman/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/traject-3.0.0/lib/traject/indexer/step.rb:135:in `each'
	 4: from /Users/jlittman/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/traject-3.0.0/lib/traject/indexer/step.rb:138:in `block in execute'
	 3: from lib/rialto/etl/configs/wos_to_sparql_statements.rb:214:in `block (2 levels) in load_config_file'
	 2: from lib/rialto/etl/configs/wos_to_sparql_statements.rb:66:in `fetch_grant_identifiers'
	 1: from lib/rialto/etl/configs/wos_to_sparql_statements.rb:66:in `reject'
lib/rialto/etl/configs/wos_to_sparql_statements.rb:66:in `block in fetch_grant_identifiers': undefined method `empty?' for 1022752:Integer (NoMethodError)
```

--- a/lib/rialto/etl/configs/wos_to_sparql_statements.rb
+++ b/lib/rialto/etl/configs/wos_to_sparql_statements.rb
@@ -63,7 +63,7 @@ end
 def fetch_grant_identifiers(json)
   Array.wrap(
     JsonPath.on(json, '$.static_data.fullrecord_metadata.fund_ack.grants.grant[*].grant_ids.grant_id').flatten
-  ).reject { |id| id.empty? || id.nil? }
+  ).reject { |id| id.to_s.empty? || id.nil? }.map(&:to_s)
 end
 
 # Return the names of the publishers

--- a/spec/configs/wos_to_sparql_statements_spec.rb
+++ b/spec/configs/wos_to_sparql_statements_spec.rb
@@ -694,4 +694,92 @@ RSpec.describe Rialto::Etl::Transformer do
       it { is_expected.to eq(['National Institute of Diabetes and Digestive and Kidney Diseases']) }
     end
   end
+  describe '#fetch_grant_identifiers' do
+    subject { indexer.fetch_grant_identifiers(json) }
+
+    let(:indexer) do
+      Traject::Indexer.new.tap do |indexer|
+        indexer.load_config_file(config_file_path)
+      end
+    end
+
+    context 'with an string grant identifiers' do
+      let(:json) do
+        <<~JSON
+          {
+          	"UID": "WOS:000359895400001",
+          	"static_data": {
+          		"fullrecord_metadata": {
+                "fund_ack": {
+                  "grants": {
+                    "count": 3,
+                    "grant": [{
+                      "grant_ids": {
+                        "grant_id": ["U01DK073983", "U01DK073983"],
+                        "count": 6
+                      }
+                    }, {
+                      "grant_ids": {
+                        "grant_id": "DK57061",
+                        "count": 1
+                      }
+                    }, {
+                      "grant_ids": {
+                        "grant_id": "PO1 DK68055",
+                        "count": 1
+                      }
+                    }]
+                  }
+                }
+          		}
+          	}
+          }
+        JSON
+      end
+
+      it {
+        is_expected.to eq(['U01DK073983', 'U01DK073983', 'DK57061', 'PO1 DK68055'])
+      }
+    end
+    context 'with an integer grant identifiers' do
+      let(:json) do
+        <<~JSON
+          {
+          	"UID": "WOS:000359895400001",
+          	"static_data": {
+          		"fullrecord_metadata": {
+                "fund_ack": {
+                  "grants": {
+                    "count": 3,
+                    "grant": [{
+                      "grant_ids": {
+                        "grant_id": [1019921, 1019922],
+                        "count": 6
+                      }
+                    }, {
+                      "grant_ids": {
+                        "grant_id": 1019923,
+                        "count": 1
+                      }
+                    }, {
+                      "grant_ids": {
+                        "grant_id": 1019924,
+                        "count": 1
+                      }
+                    }]
+                  }
+                }
+          		}
+          	}
+          }
+        JSON
+      end
+
+      it {
+        # rubocop:disable Style/WordArray
+        is_expected.to eq(['1019921', '1019922', '1019923', '1019924'])
+        # rubocop:enable Style/WordArray
+      }
+    end
+  end
 end

